src/App.tsx

```diff
@@ -15,36 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+function prepareGoods(
+  goods: string[],
+  sortedField: string,
+  isReversed: boolean,
+) {
+  const copyGoods = [...goods];
+
+  if (sortedField !== SortType.default) {
+    copyGoods.sort((elementA, elementB) => {
+      switch (sortedField) {
+        case SortType.alphabet:
+          return elementA.localeCompare(elementB);
+
+        case SortType.length:
+          return elementA.length - elementB.length;
+
+        case SortType.default:
+        default:
+          return 0;
```

Comment:
Pay attention that for the default case sort method would be called without any sorting, which is not good. 
Consider not calling sort method at all if it's the default case